src/modules/campaign-monitor/services/campaign-monitor.service.ts

```diff
@@ -181,4 +268,23 @@ export class CampaignMonitorService {
 			end_time: visit.endAt ? formatAsBelgianDate(visit.endAt, 'HH:mm') : '',
 		};
 	}
+
+	public convertPreferencesToNewsletterTemplateData(
+		user: SessionUserEntity,
+		optin_mail_lists: string,
+		resubscribe: boolean
+	) {
+		return {
+			Name: user.getFullName(),
+			EmailAddress: user.getMail(),
+			optin_mail_lists,
+			gebruikersgroep: user.getGroupId(),
+			is_sleutel_gebruiker: user.getIsKeyUser(),
+			firstname: user.getFirstName(),
+			lastname: user.getLastName(),
+			aangemaakt_op: new Date(),
+			Resubscribe: resubscribe,
+			ConsentToTrack: resubscribe ? 'Yes' : 'Unchanged',
```

Comment:
This requires 2 more fields according to the confluence doc: https://meemoo.atlassian.net/wiki/spaces/AVO2/pages/3830644751/Hetarchief+CM+Campaign+Monitor+lijst+nieuwsbrief
* laatst_ingelogd_op: zit in de database als: profile.last_access_at
* organisatie => zit in de database als profile.organisation_schema_identifier en mogelijks ook in het user object